include/nanorange/algorithm/partition.hpp

```diff
@@ -16,12 +19,13 @@ namespace detail {
 struct partition_fn {
 private:
     template <typename I, typename S, typename Pred, typename Proj>
-    static constexpr I impl(I first, S last, Pred& pred, Proj& proj)
+    static constexpr subrange<I> impl(I first, S last, Pred& pred, Proj& proj)
     {
+	I first_copy = first;
         first = nano::find_if_not(std::move(first), last, pred, proj);
```

Comment:
Rather than creating `first_copy`, you can do something like
```cpp
auto iter = nano::find_if_not(first, last, pred, proj);
```

Reply:
That definitely looks nicer.